test/files-helper.js

```diff
@@ -71,30 +71,44 @@ FilesHelper.prototype.after = function() {
   return removeItems([this.dirs.lockDir, this.dirs.tempDir], 'rmdir');
 };
 
-FilesHelper.prototype.createRepoDir = function(done) {
+FilesHelper.prototype.createRepoDir = function() {
   var helper = this;
-  fs.mkdir(this.dirs.testRepoDir, '0700', function() {
-    fs.writeFile(helper.files.configYml, '', done);
+
+  return new Promise(function(resolve, reject) {
+    fs.mkdir(helper.dirs.testRepoDir, '0700', function(err) {
+      if (err) {
+        return reject(err);
+      }
+      fs.writeFile(helper.files.configYml, '', function(err) {
+        if (err) {
+          return reject(err);
+        }
+        resolve();
+      });
+    });
   });
 };
 
-FilesHelper.prototype.createRepoWithFiles = function(nameToContents, done) {
+FilesHelper.prototype.createRepoWithFiles = function(nameToContents) {
   var helper = this,
-      filesRemaining,
-      allDone;
+      writeFile;
 
   this.filesToDelete = Object.keys(nameToContents);
-  filesRemaining = this.filesToDelete.length,
-  allDone = function() {
-    filesRemaining--;
-    if (filesRemaining === 0) { done(); }
+  writeFile = function(filename) {
+    return new Promise(function(resolve, reject) {
+      fs.writeFile(filename, nameToContents[filename], function(err) {
+        if (err) {
+          return reject(err);
+        }
+        resolve();
+      });
+    });
   };
 
-  this.createRepoDir(function() {
-    helper.filesToDelete.map(function(name) {
-      fs.writeFile(name, nameToContents[name], allDone);
+  return this.createRepoDir()
+    .then(function() {
```

Comment:
So here's the thing: I could add a `catch` block here, perhaps to log the error, but it currently won't propagate to Mocha in any substantial way. After this PR and #33 go in, I'll cherry-pick 5332f2cec9d441bc3a28475eefa7c9fb6aa652dc from the [`site-builder-refactor` branch](https://github.com/18F/pages-server/commits/site-builder-refactor) to create a new PR.

In that commit, `SiteBuilder.build` returns a `Promise`, and the tests are `Promise`-chained all the way so that any error from this `Promise` is propagated to Mocha.